flake.nix

```diff
@@ -1,206 +1,106 @@
-        style = with nixpkgsCleanFor system;
-          mkShell {
-            nativeBuildInputs = [
-              (haskell.lib.justStaticExecutables
-               (ghcVersion pkgs).fourmolu)
-              (haskell.lib.justStaticExecutables
-               (ghcVersion pkgs).hlint)
-              pkgs.hpack
-            ];
-            shellHook = ''
-              hpack booster && hpack dev-tools
-            '';
+          pkgs = import nixpkgs {
+            inherit system;
+            overlays = [ self.overlays.haskell-versions self.overlays.default ];
+          };
+        in
+        {
+          packages = rec {
+            default = hs-backend-booster;
+            kore-rpc-types = pkgs.haskell.packages.${ghcVer}.kore-rpc-types;
+            kore = pkgs.haskell.packages.${ghcVer}.kore;
+            hs-backend-booster = pkgs.haskell.packages.${ghcVer}.hs-backend-booster;
+            hs-backend-booster-dev-tools = pkgs.haskell.packages.${ghcVer}.hs-backend-booster-dev-tools;
           };
-        cabal = let pkgs = nixpkgsFor system;
-        in pkgs.haskell-backend.pkgSet.shellFor {
-          packages = pkgs.haskell-backend.localPkgsSelector;
-          nativeBuildInputs = [
-              (ghcVersion pkgs).cabal-install
-              pkgs.hpack
-              pkgs.jq
-              pkgs.nix
-              pkgs.z3
-              pkgs.lsof
-          ];
-          shellHook = ''
-            hpack booster && hpack dev-tools
-          '';
+
+          devShells.default =
+            let haskellPackages = pkgs.haskell.packages.${ghcVer};
+            in
+            haskellPackages.shellFor {
+              packages = p: [ self.packages.${system}.sample ];
+              withHoogle = false;
+              buildInputs = with haskellPackages; [
+                cabal-install
+              ];
+            };
         };
-      });
+    in
+    flake-utils.lib.eachDefaultSystem out // {
+      overlays = {
+        haskell-versions = makeHaskellOverlay (final: prev:
+          import ./nix/some-cabal-hashes.nix {
+            self = final;
 
-      devShell =
-        perSystem (system: (nixpkgsFor system).haskell-backend.devShell);
+            overrides = 
+              let 
+                hprev = prev.haskell.packages;
+                hfinal = final.haskell.packages;
+              in {
+                # FIXME these two overrides cause an error in some-cabal-hashes.nix
+                # tasty-test-reporter = 
+                #     final.fetchFromGitHub {
+                #       owner = "goodlyrottenapple";
+                #       repo = "tasty-test-reporter";
+                #       rev = "b704130545aa3925a8487bd3e92f1dd5ce0512e2";
+                #       sha256 = "sha256-9CN59Im0BC3vSVhL85v5eXPYYoPbV3NAuv893tXpr/U=";
+                #     };
+                # profiteur =
+                #     final.fetchFromGitHub {
+                #       owner = "goodlyrottenapple";
+                #       repo = "profiteur";
```

Comment:
These two are custom dependencies from github (forks of repositories that cause dependency issues). The code here follows the `some-cabal-hashes` example but is seemingly ignored. I already removed all code that uses `tasty-test-reporter`, we can remove the `profiteur` consumer, too and forget about this issue.